pkg/db/client.go

```diff
@@ -146,6 +169,57 @@ func (c *ClientSqlx) Exec(ctx context.Context, query string, args ...interface{}
 	return c.db.ExecContext(ctx, query, args...)
 }
 
+func (c *ClientSqlx) ExecMultiInTx(ctx context.Context, sqllers ...Sqller) (results []sql.Result, err error) {
+	var tx *sql.Tx
+	var res sql.Result
+	var buildErr error
+	var qry string
+	var queries []string
+	var args []interface{}
+	var argss [][]interface{}
+
+	for i, sqller := range sqllers {
+		if qry, args, buildErr = sqller.ToSql(); buildErr != nil {
+			return nil, fmt.Errorf("can not build sql #%d: %w", i, err)
+		}
+
+		queries = append(queries, qry)
+		argss = append(argss, args)
+	}
+
+	if tx, err = c.BeginTx(ctx, &sql.TxOptions{}); err != nil {
+		err = fmt.Errorf("can not begin transaction: %w", err)
+		return
+	}
+
+	defer func() {
+		if err == nil {
+			return
+		}
+
+		if errRollback := tx.Rollback(); errRollback != nil {
+			err = multierror.Append(err, fmt.Errorf("can not roolback tx: %w", errRollback))
```

Comment:
Do we need to do it like this instead? `err = multierror.Append(err, fmt.Errorf("can not roolback tx: %w", errRollback)).ErrorOrNil()`

Reply:
applied